libs/sdk-common/src/input_parser.rs

```diff
@@ -183,8 +191,21 @@ use crate::prelude::*;
 pub async fn parse(
     input: &str,
     external_input_parsers: Option<&[ExternalInputParser]>,
+    dns_resolver: Option<&AsyncResolver<GenericConnector<TokioRuntimeProvider>>>,
 ) -> Result<InputType> {
-    let input = input.trim();
+    let mut input = input.trim();
+
+    // Try to parse the destination as a bip353 address.
+    let input_str = if let Some(resolver) = dns_resolver {
```

Comment:
NIT: `str` in a name leads one to believe it's a string slice (`&str`) but in this case it's a `String`